src/Vault.sol

```diff
@@ -253,17 +254,31 @@ contract Vault {
             _setBaseCurrency(baseCurrency);
         }
         success = getFreeMargin() >= amount;
+        // Update the vault values
+        (address[] memory AllAssetAddresses, uint256[] memory AllAssetIds, uint256[] memory AllAssetAmounts) =
```

Comment:
Can we rename this to ` (address[] memory assetAddresses, uint256[] memory assetIds, uint256[] memory assetAmounts)` to be consistent with other generateAssetData's?